### PR TITLE
[gendump] Add Support to dump BCM-DNX commands

### PR DESCRIPTION
**Pull request by saksarav-nokia**
In generate_dump script, the function collect_broadcom tries to dump BCM XGS commands
even in DNX switchs and error message is printed, hence generate_dump command fails.
Modified the collect_broadcom function to check for BCM family and dump
the DNX commands if the family is DNX.

Signed-off-by: Sakthivadivu Saravanaraj <sakthivadivu.saravanaraj@nokia.com>

#### What I did
Modified the generate_dump script to dump the corresponding Broadcom diag shell commands based on Broadcom asic family. 
#### How I did it
Modified the collect_broadcom function in generate_dump to get the Broadcom asic family by extracting the "version" command output as per Broadcom's suggestion and dump the Broadcom DNX commands if the Broadcom asic is DNX family. 
If it is not DNX family, dump the existing XGS commands. 
#### How to verify it
Ran sonic cli command "sudo generate_dump " in DNX platform and XGS platform and verified that the tgz files are created with all the information dumped by generate_dump script.

### CODE REVIEW

**Comment by saksarav-nokia** [comment]
@judyjoseph, Azure pipeline fails which are the failing tests unrelated to my changes. Could you please restart the pipeline?
Thanks,
Sakthi

**Review by vmittal-msft — COMMENTED** [review]
Does this command dump the "set of commands" into tar file from single asic or multiple asics ?

**Review by arlakshm — COMMENTED** [review]
as comments

**Comment by saksarav-nokia** [comment]
> Does this command dump the "set of commands" into tar file from single asic or multiple asics ?
 @vmittal-msft , All the commands are collected from all the asics  . Please check save_bcmcmd_all_ns

File: scripts/generate_dump [hunk 1]
```diff
@@ -985,28 +985,61 @@ collect_broadcom() {
...
+       save_bcmcmd_all_ns "\"dbal table dump table=L3_MY_MAC_DA_PREFIXES\"" "l3.mymac.summary"
+       save_bcmcmd_all_ns "\"dbal table dump table=INGRESS_VLAN_MEMBERSHIP\"" "ing.vlan.summary"
+       save_bcmcmd_all_ns "\"dbal table dump table=LOCAL_SBC_IN_LIF_MATCH_INFO_SW\"" "sbc.inlif.summary"
+       save_bcmcmd_all_ns "\"dbal table dump table=SNIF_COMMAND_TABLE\"" "snif.command.summary"
+       save_bcmcmd_all_ns "\"port mgmt dump full\"" "port.mgmt.summary"
+       save_bcmcmd_all_ns "\"tm lag\"" "tm.lag.summary"
+       save_bcmcmd_all_ns "\"pp info fec\"" "pp.fec.summary"
+       save_bcmcmd_all_ns "\"nif sts\"" "nif.sts.summary"
+       save_bcmcmd_all_ns "\"port pm info\"" "port.pm.summary"
+       save_bcmcmd_all_ns "\"conf show\"" "conf.show.summary"
```
[the remark this answers is not in the document]
Comment (saksarav-nokia) [reply scripts/generate_dump:1018]: @arlakshm , i will add these commands and update the PR

**Comment by saksarav-nokia** [comment]
@vmittal-msft @arlakshm , i have addressed you review comments. Please take a look at it.

**Review by vmittal-msft — APPROVED** [review]

**Review by arlakshm — APPROVED** [review]

**Pull request merged**